Assets/Scripts/R&D Tree/RDTreeManager.cs

```diff
@@ -0,0 +1,115 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class RDTreeManager : MonoBehaviour
+{
+    [SerializeField] private List<ResearchNode> _firstAvailableNodes = new List<ResearchNode>();
+
+    public ResearchNode CurrentResearching { get; private set; }
+    public List<ResearchNode> AllNodes { get; private set; }
+    public List<ResearchNode> ResearchQueue { get; private set; }
+
+    private void Awake()
+    {
+        AllNodes = new List<ResearchNode>();
+        ResearchQueue = new List<ResearchNode>();
+    }
+
+    private void Start() => ActivateFirstNodes();
+
+    /// <summary>
+    /// This function sets the firstavailable nodes in the tree active and available
+    /// </summary>
+    private void ActivateFirstNodes()
+    {
+        foreach (ResearchNode node in _firstAvailableNodes)
+        {
+            node.SetResearchStatus(ResearchNode.ResearchStates.available);
+        }
+    }
+
+    public void AddNodeToList(ResearchNode node) => AllNodes.Add(node);
+
+    /// <summary>
+    /// Function that returns a list of all the nodes that are available to buy
+    /// </summary>
+    /// <returns>Return a list with all the available nodes</returns>
+    public List<ResearchNode> AvailableNodes()
+    {
+        List<ResearchNode> availableNodes = new List<ResearchNode>();
+
+        foreach (ResearchNode node in AllNodes)
+        {
+            if (node.CurrentResearchState == ResearchNode.ResearchStates.available)
+                availableNodes.Add(node);
+        }
+
+        return availableNodes;
+    }
+
+    /// <summary>
+    /// Checks if there is an node in the queue and returns that node
+    /// </summary>
+    /// <returns>Returns the next node in the queue</returns>
+    public ResearchNode NextInQueue()
+    {
+        if (ResearchQueue.Count > 0)
+            return ResearchQueue[0];
+        else
+            return null;
+    }
+
+    /// <summary>
+    /// This method will pause the research of the one that is currently researching and then it will set the new research.
```

Comment:
Be consistent with the use of periods at the end of sentences. Not important, but keep an eye out for it